### PR TITLE
Add missing smart wallet (PDA wallet) support

### PR DESCRIPTION
**Pull request by valentinmadrid**
This adds allowOwnerOffCurve to ATA derivations for Realms to be compatible with the SquadsX extension.

### CODE REVIEW

File: components/GovernancePower/Vanilla/useDepositCallback.tsx [hunk 1]
```diff
@@ -38,24 +38,53 @@ export const useDepositCallback = (
...
+          wallet!.publicKey!,
+          amount
+        ) 
 
       signers.push(transferAuthority)
+      }
 
       const programVersion = await fetchProgramVersion(connection, realm.owner)
 
+      if (wallet?.name == "SquadsX") {
```
Comment (asktree) [inline components/GovernancePower/Vanilla/useDepositCallback.tsx:74]: could you explain this and why a special case is needed? and if it really is needed, please leave comments in the code about it

Comment (valentinmadrid) [reply components/GovernancePower/Vanilla/useDepositCallback.tsx:77]: Added comments.

File: components/GovernancePower/Vanilla/useDepositCallback.tsx [hunk 2]
```diff
@@ -38,36 +38,70 @@ export const useDepositCallback = (
...
-        instructions,
-        [],
-        userAtaPk,
-        wallet!.publicKey!,
-        amount
-      )
+      let transferAuthority;
+
+      // Checks if the connected wallet is the Squads Multisig extension. This wallet needs custom logic as ephemeral Keypair signatures can not be used after the transaction blockhash has expired.
+      if (wallet?.name == "SquadsX") {
```
Comment (asktree) [inline components/GovernancePower/Vanilla/useDepositCallback.tsx:51]: style: just use a ternary please!

File: components/GovernancePower/Vanilla/useDepositCallback.tsx [hunk 3]
```diff
@@ -64,9 +70,9 @@ export const useDepositCallback = (
         userAtaPk,
         mint,
         walletPk,
-        transferAuthority.publicKey,
+        publicKeyToUse,
         walletPk,
-        amount
+        amount,
```
Comment (asktree) [inline components/GovernancePower/Vanilla/useDepositCallback.tsx:75]: should this comma be here? seems odd

Comment (valentinmadrid) [reply components/GovernancePower/Vanilla/useDepositCallback.tsx:75]: My formatter uses trailing commas, if not wanted I can remove it.

Comment (asktree) [reply components/GovernancePower/Vanilla/useDepositCallback.tsx]: huh. surprised that ours doesn't appear to be opinionated on it.

**Pull request merged**